packages/iris-grid/src/sidebar/visibility-ordering-builder/VisibilityOrderingBuilder.tsx

```diff
@@ -635,11 +635,47 @@ class VisibilityOrderingBuilder extends PureComponent<
     option: keyof typeof VisibilityOrderingBuilder.SORTING_OPTIONS
   ): void {
     const { model, onMovedColumnsChanged } = this.props;
+    const tree = this.getTreeItems();
+    const firstIndex = this.getFirstMovableIndex() ?? 0;
+    const lastIndex = this.getLastMovableIndex() ?? tree.length - 1;
+    const moveableTree = tree.slice(firstIndex, lastIndex + 1);
+
+    // add frozen moves
+    const initialAndFrozenMovedColumns = [...model.initialMovedColumns];
+    for (let i = 0; i < model.frozenColumns.length; i += 1) {
+      const frozenColumn = model.frozenColumns[i];
+      // get index of frozenColumn
+      let frozenIndex = 0;
+      for (let j = 0; j < model.columns.length; j += 1) {
+        if (model.columns[j].name === frozenColumn) {
+          frozenIndex = j;
+          break;
+        }
+      }
+      // offset by move changes
+      for (let j = 0; j < initialAndFrozenMovedColumns.length; j += 1) {
+        const { from, to } = initialAndFrozenMovedColumns[j];
+        // taken from in front, decrease index
+        if (Array.isArray(from)) {
+          if (from[0] < frozenIndex) frozenIndex -= 1;
+        } else if (from < frozenIndex) {
+          frozenIndex -= 1;
+        }
+        // placed in front, increase index
+        if (to < frozenIndex) frozenIndex += 1;
+      }
+      if (frozenIndex !== i) {
+        initialAndFrozenMovedColumns.push({
+          from: frozenIndex,
+          to: i,
+        });
+      }
```

Comment:
You can use `GridUtils.getVisibleIndex(frozenIndex, initialAndFrozenMovedColumns)` instead of re-creating the algorithm.